--- a/com/hypixel/hytale/builtin/npccombatactionevaluator/evaluator/CombatActionEvaluatorConfig.java
+++ b/com/hypixel/hytale/builtin/npccombatactionevaluator/evaluator/CombatActionEvaluatorConfig.java
@@ -30,7 +_,7 @@
       .addValidator(CombatActionOption.VALIDATOR_CACHE.getMapValueValidator())
       .documentation("A map of all available combat actions this NPC can take.")
       .add()
-      .<Map>append(
+      .append(
          new KeyedCodec<>("ActionSets", new MapCodec<>(CombatActionEvaluatorConfig.ActionSet.CODEC, Object2ObjectOpenHashMap::new)),
          (option, o) -> option.actionSets = o,
          option -> option.actionSets
@@ -208,7 +_,7 @@
          .addValidator(Validators.weaklyMonotonicSequentialDoubleArrayValidator())
          .documentation("A random range to pick a cooldown between basic attacks from.")
          .add()
-         .<Map>appendInherited(
+         .appendInherited(
             new KeyedCodec<>("InteractionVars", new MapCodec<>(RootInteraction.CHILD_ASSET_CODEC, Object2ObjectOpenHashMap::new)),
             (basicAttacks, v) -> basicAttacks.interactionVars = v,
             basicAttacks -> basicAttacks.interactionVars,
