--- a/com/hypixel/hytale/builtin/buildertools/BuilderToolsPlugin.java
+++ b/com/hypixel/hytale/builtin/buildertools/BuilderToolsPlugin.java
@@ -385,22 +_,22 @@
       this.prefabEditSessionManager = new PrefabEditSessionManager(this);
       this.prefabEditSessionResourceType = entityStoreRegistry.registerResource(PrefabEditSession.class, "PrefabEditSession", PrefabEditSession.CODEC);
       AssetRegistry.register(
-         ((HytaleAssetStore.Builder)((HytaleAssetStore.Builder)((HytaleAssetStore.Builder)HytaleAssetStore.builder(
-                        PrefabEditorCreationSettings.class, new DefaultAssetMap()
+         HytaleAssetStore.builder(
+                        PrefabEditorCreationSettings.class, new DefaultAssetMap<>()
                      )
-                     .setPath("PrefabEditorCreationSettings"))
-                  .setKeyFunction(PrefabEditorCreationSettings::getId))
-               .setCodec(PrefabEditorCreationSettings.CODEC))
-            .build()
+                     .setPath("PrefabEditorCreationSettings")
+                     .setKeyFunction(PrefabEditorCreationSettings::getId)
+                     .setCodec(PrefabEditorCreationSettings.CODEC)
+                     .build()
       );
       AssetRegistry.register(
-         ((HytaleAssetStore.Builder)((HytaleAssetStore.Builder)((HytaleAssetStore.Builder)HytaleAssetStore.builder(
-                        ScriptedBrushAsset.class, new DefaultAssetMap()
+         HytaleAssetStore.builder(
+                        ScriptedBrushAsset.class, new DefaultAssetMap<>()
                      )
-                     .setPath("ScriptedBrushes"))
-                  .setKeyFunction(ScriptedBrushAsset::getId))
-               .setCodec(ScriptedBrushAsset.CODEC))
-            .build()
+                     .setPath("ScriptedBrushes")
+                     .setKeyFunction(ScriptedBrushAsset::getId)
+                     .setCodec(ScriptedBrushAsset.CODEC)
+                     .build()
       );
       commandRegistry.registerCommand(new ClearBlocksCommand());
       commandRegistry.registerCommand(new ClearEntitiesCommand());
@@ -2268,11 +_,12 @@
                      if (entities) {
                         Store<EntityStore> store = world.getEntityStore().getStore();
                         ArrayList<Ref<EntityStore>> entitiesToRemove = cut ? new ArrayList<>() : null;
+                        List<SelectionSnapshot<?>> snapshotsFinal = snapshots;
                         BuilderToolsPlugin.forEachCopyableInSelection(world, xMin, yMin, zMin, width, height, depth, e -> {
                            Holder<EntityStore> holder = store.copyEntity(e);
                            this.selection.addEntityFromWorld(holder);
                            if (cut) {
-                              snapshots.add(new EntityRemoveSnapshot(e));
+                              snapshotsFinal.add(new EntityRemoveSnapshot(e));
                               entitiesToRemove.add(e);
                            }
                         });
@@ -2487,7 +_,7 @@
                            rotatedLocation.x + x,
                            rotatedLocation.y + y,
                            rotatedLocation.z + z,
-                           blockChange.block,
+                           blockChangex.block,
                            newRotation,
                            FillerBlockUtil.pack(x, y, z),
                            0,
