--- a/com/hypixel/hytale/assetstore/codec/ContainedAssetCodec.java
+++ b/com/hypixel/hytale/assetstore/codec/ContainedAssetCodec.java
@@ -55,7 +_,7 @@
    @Nullable
    @Override
    public K decode(@Nonnull BsonValue bsonValue, ExtraInfo extraInfo) {
-      if (!(extraInfo instanceof AssetExtraInfo<K> assetExtraInfo)) {
+      if (!(extraInfo instanceof AssetExtraInfo assetExtraInfo)) {
          throw new UnsupportedOperationException("Unable to decode asset from codec used outside of an AssetStore");
       } else if (bsonValue.isString()) {
          return this.codec.getKeyCodec().getChildCodec().decode(bsonValue, extraInfo);
@@ -66,7 +_,7 @@
          K id;
          switch (this.mode) {
             case GENERATE_ID: {
-               id = this.keyGenerator.apply(assetExtraInfo);
+               id = (K) this.keyGenerator.apply(assetExtraInfo);
                boolean inheritContainerTags = false;
                break;
             }
@@ -88,9 +_,9 @@
                break;
             }
             case INJECT_PARENT: {
-               id = this.keyGenerator.apply(assetExtraInfo);
+               id = (K) this.keyGenerator.apply(assetExtraInfo);
                if (parentId == null && !assetExtraInfo.getKey().equals(id)) {
-                  parentId = assetExtraInfo.getKey();
+                  parentId = (K) assetExtraInfo.getKey();
                }
 
                boolean inheritContainerTags = true;
@@ -126,7 +_,7 @@
    @Nullable
    @Override
    public K decodeJson(@Nonnull RawJsonReader reader, ExtraInfo extraInfo) throws IOException {
-      if (!(extraInfo instanceof AssetExtraInfo<K> assetExtraInfo)) {
+      if (!(extraInfo instanceof AssetExtraInfo assetExtraInfo)) {
          throw new UnsupportedOperationException("Unable to decode asset from codec used outside of an AssetStore");
       } else {
          int lineStart = reader.getLine() - 1;
@@ -146,7 +_,7 @@
             K id;
             switch (this.mode) {
                case GENERATE_ID: {
-                  id = this.keyGenerator.apply(assetExtraInfo);
+                  id = (K) this.keyGenerator.apply(assetExtraInfo);
                   boolean inheritContainerTags = false;
                   break;
                }
@@ -168,9 +_,9 @@
                   break;
                }
                case INJECT_PARENT: {
-                  id = this.keyGenerator.apply(assetExtraInfo);
+                  id = (K) this.keyGenerator.apply(assetExtraInfo);
                   if (parentId == null && !assetExtraInfo.getKey().equals(id)) {
-                     parentId = assetExtraInfo.getKey();
+                     parentId = (K) assetExtraInfo.getKey();
                   }
 
                   boolean inheritContainerTags = true;
