--- a/com/hypixel/hytale/builtin/adventure/objectives/ObjectivePlugin.java
+++ b/com/hypixel/hytale/builtin/adventure/objectives/ObjectivePlugin.java
@@ -145,43 +_,35 @@
    protected void setup() {
       instance = this;
       AssetRegistry.register(
-         ((HytaleAssetStore.Builder)((HytaleAssetStore.Builder)((HytaleAssetStore.Builder)((HytaleAssetStore.Builder)HytaleAssetStore.builder(
-                           ObjectiveAsset.class, new DefaultAssetMap()
-                        )
-                        .setPath("Objective/Objectives"))
-                     .setCodec(ObjectiveAsset.CODEC))
-                  .setKeyFunction(ObjectiveAsset::getId))
-               .loadsAfter(ItemDropList.class, Item.class, BlockType.class, ReachLocationMarkerAsset.class))
-            .build()
-      );
-      AssetRegistry.register(
-         ((HytaleAssetStore.Builder)((HytaleAssetStore.Builder)((HytaleAssetStore.Builder)((HytaleAssetStore.Builder)((HytaleAssetStore.Builder)HytaleAssetStore.builder(
-                              ObjectiveLineAsset.class, new DefaultAssetMap()
-                           )
-                           .setPath("Objective/ObjectiveLines"))
-                        .setCodec(ObjectiveLineAsset.CODEC))
-                     .setKeyFunction(ObjectiveLineAsset::getId))
-                  .loadsAfter(ObjectiveAsset.class))
-               .loadsBefore(GameplayConfig.class))
-            .build()
-      );
-      AssetRegistry.register(
-         ((HytaleAssetStore.Builder)((HytaleAssetStore.Builder)((HytaleAssetStore.Builder)((HytaleAssetStore.Builder)HytaleAssetStore.builder(
-                           ObjectiveLocationMarkerAsset.class, new DefaultAssetMap()
-                        )
-                        .setPath("Objective/ObjectiveLocationMarkers"))
-                     .setCodec(ObjectiveLocationMarkerAsset.CODEC))
-                  .setKeyFunction(ObjectiveLocationMarkerAsset::getId))
-               .loadsAfter(ObjectiveAsset.class, Environment.class, Weather.class))
-            .build()
-      );
-      AssetRegistry.register(
-         ((HytaleAssetStore.Builder)((HytaleAssetStore.Builder)((HytaleAssetStore.Builder)HytaleAssetStore.builder(
-                        ReachLocationMarkerAsset.class, new DefaultAssetMap()
-                     )
-                     .setPath("Objective/ReachLocationMarkers"))
-                  .setCodec(ReachLocationMarkerAsset.CODEC))
-               .setKeyFunction(ReachLocationMarkerAsset::getId))
+         HytaleAssetStore.builder(ObjectiveAsset.class, new DefaultAssetMap<>())
+            .setPath("Objective/Objectives")
+            .setCodec(ObjectiveAsset.CODEC)
+            .setKeyFunction(ObjectiveAsset::getId)
+            .loadsAfter(ItemDropList.class, Item.class, BlockType.class, ReachLocationMarkerAsset.class)
+            .build()
+      );
+      AssetRegistry.register(
+         HytaleAssetStore.builder(ObjectiveLineAsset.class, new DefaultAssetMap<>())
+            .setPath("Objective/ObjectiveLines")
+            .setCodec(ObjectiveLineAsset.CODEC)
+            .setKeyFunction(ObjectiveLineAsset::getId)
+            .loadsAfter(ObjectiveAsset.class)
+            .loadsBefore(GameplayConfig.class)
+            .build()
+      );
+      AssetRegistry.register(
+         HytaleAssetStore.builder(ObjectiveLocationMarkerAsset.class, new DefaultAssetMap<>())
+            .setPath("Objective/ObjectiveLocationMarkers")
+            .setCodec(ObjectiveLocationMarkerAsset.CODEC)
+            .setKeyFunction(ObjectiveLocationMarkerAsset::getId)
+            .loadsAfter(ObjectiveAsset.class, Environment.class, Weather.class)
+            .build()
+      );
+      AssetRegistry.register(
+         HytaleAssetStore.builder(ReachLocationMarkerAsset.class, new DefaultAssetMap<>())
+            .setPath("Objective/ReachLocationMarkers")
+            .setCodec(ReachLocationMarkerAsset.CODEC)
+            .setKeyFunction(ReachLocationMarkerAsset::getId)
             .build()
       );
       this.objectiveDataStore = new ObjectiveDataStore(this.config.get().getDataStoreProvider().create(Objective.CODEC));
@@ -307,7 +_,7 @@
    ) {
       ObjectiveTaskAsset.CODEC.register(id, assetClass, assetCodec);
       ObjectiveTask.CODEC.register(id, implementationClass, implementationCodec);
-      this.taskGenerators.put(assetClass, generator);
+      this.taskGenerators.put(assetClass, (TriFunction<ObjectiveTaskAsset, Integer, Integer, ? extends ObjectiveTask>) generator);
       this.objectiveDataStore.registerTaskRef(implementationClass);
    }
 
@@ -315,7 +_,7 @@
       String id, Class<T> assetClass, Codec<T> codec, Function<T, U> generator
    ) {
       ObjectiveCompletionAsset.CODEC.register(id, assetClass, codec);
-      this.completionGenerators.put(assetClass, generator);
+      this.completionGenerators.put(assetClass, (Function<ObjectiveCompletionAsset, ? extends ObjectiveCompletion>) generator);
    }
 
    public ObjectiveTask createTask(@Nonnull ObjectiveTaskAsset task, int taskSetIndex, int taskIndex) {
