--- a/com/hypixel/hytale/server/core/plugin/PluginBase.java
+++ b/com/hypixel/hytale/server/core/plugin/PluginBase.java
@@ -42,7 +_,7 @@
 
 public abstract class PluginBase implements CommandOwner {
    @Nonnull
-   public static final MetricsRegistry<PluginBase> METRICS_REGISTRY = new MetricsRegistry<>(MetricProvider.maybe(Function.identity()))
+   public static final MetricsRegistry<PluginBase> METRICS_REGISTRY = new MetricsRegistry<PluginBase>(MetricProvider.maybe(Function.identity()))
       .register("Identifier", plugin -> plugin.identifier.toString(), Codec.STRING)
       .register("Type", PluginBase::getType, new EnumCodec<>(PluginType.class))
       .register("Manifest", plugin -> plugin.manifest, PluginManifest.CODEC)
