--- a/com/hypixel/hytale/server/core/command/system/AbstractCommand.java
+++ b/com/hypixel/hytale/server/core/command/system/AbstractCommand.java
@@ -589,7 +_,7 @@
          } else {
             commandContext.appendArgumentData(
                optionalArg,
-               optionalArgValues.isEmpty() ? EMPTY_STRING_ARRAY : ((List)optionalArgValues.getFirst()).toArray(EMPTY_STRING_ARRAY),
+               optionalArgValues.isEmpty() ? EMPTY_STRING_ARRAY : (optionalArgValues.getFirst()).toArray(EMPTY_STRING_ARRAY),
                false,
                parseResult
             );
